core/src/main/java/org/pentaho/di/core/row/ValueDataUtil.java

```diff
@@ -0,0 +1,2255 @@
+          return valueB;
+        } else {
+          return Boolean.valueOf( valueA.booleanValue() || valueB.booleanValue() );
+        }
+      }
+      case ValueMetaInterface.TYPE_BIGNUMBER: {
+        BigDecimal valueA = metaA.getBigNumber( dataA );
+        BigDecimal valueB = metaB.getBigNumber( dataB );
+        if ( valueB == null ) {
+          return valueA;
+        } else if ( valueA == null ) {
+          return valueB;
+        } else {
+          return valueA.add( valueB );
+        }
+      }
+      default:
+        throw new KettleValueException( "The 'plus' function only works on numeric data and Strings." );
+    }
+  }
+
+  public static Object plus3( ValueMetaInterface metaA, Object dataA, ValueMetaInterface metaB, Object dataB,
+    ValueMetaInterface metaC, Object dataC ) throws KettleValueException {
+    if ( dataA == null || dataB == null || dataC == null ) {
+      return null;
+    }
+
+    switch ( metaA.getType() ) {
+      case ValueMetaInterface.TYPE_STRING:
+        return metaA.getString( dataA ) + metaB.getString( dataB ) + metaC.getString( dataC );
+      case ValueMetaInterface.TYPE_NUMBER:
+        return new Double( metaA.getNumber( dataA ).doubleValue()
+          + metaB.getNumber( dataB ).doubleValue() + metaC.getNumber( dataC ).doubleValue() );
+      case ValueMetaInterface.TYPE_INTEGER:
+        return new Long( metaA.getInteger( dataA ).longValue()
+          + metaB.getInteger( dataB ).longValue() + metaC.getInteger( dataC ).longValue() );
+      case ValueMetaInterface.TYPE_BOOLEAN:
+        return Boolean.valueOf( metaA.getBoolean( dataA ).booleanValue()
+          || metaB.getBoolean( dataB ).booleanValue() || metaB.getBoolean( dataC ).booleanValue() );
+      case ValueMetaInterface.TYPE_BIGNUMBER:
+        return metaA.getBigNumber( dataA ).add( metaB.getBigNumber( dataB ).add( metaC.getBigNumber( dataC ) ) );
+
+      default:
+        throw new KettleValueException( "The 'plus' function only works on numeric data and Strings." );
+    }
+  }
+
+  public static Object sum( ValueMetaInterface metaA, Object dataA, ValueMetaInterface metaB, Object dataB ) throws KettleValueException {
+    if ( dataA == null && dataB == null ) {
+      return null;
+    }
+    if ( dataA == null && dataB != null ) {
+      Object value = metaA.convertData( metaB, dataB );
+      metaA.setStorageType( ValueMetaInterface.STORAGE_TYPE_NORMAL );
+      return value;
+    }
+    if ( dataA != null && dataB == null ) {
+      return dataA;
+    }
+
+    return plus( metaA, dataA, metaB, dataB );
+  }
+
+  /**
+   * @deprecated Use {@link ValueDataUtil#loadFileContentInBinary(ValueMetaInterface, Object, boolean)} instead
+   */
+  @Deprecated
+  public static Object loadFileContentInBinary( ValueMetaInterface metaA, Object dataA ) throws KettleValueException {
+    Object content = null;
+    try {
+      content = loadFileContentInBinary( metaA, dataA, true );
+    } catch ( KettleFileNotFoundException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)